### DISCUSSION TITLE
mutationFn parameter in useMutation

### DISCUSSION THREAD

**Discussion by anothertempore** [TanStack/query#1070] [General] [answered]
Hi there 👋

Firstly thanks for your amazing library!

In my scenario, every mutationFns are quite similar, they almost post requests with different payload and api path. 

So I'm wondering do we consider adding a global mutationFn like queryFn, I only need to pass the api path to useMutation. or what can I do to avoid repeating code?

like this:
```
const [mutate, statusInfo] = useMutation('/api/user')

mutate({ name: 'tom' })
```

Thanks!

**Comment by TkDodo** [comment, marked as answer]
I don't see that much overhead / repeating code in your example, but if you want, most abstractions can be handled with custom hooks:

```
export const useMyMutation = apiPath => useMutation(payload => axios.post(apiPath, payload))
```

if error handling is all the same, you can do that in that hook. usage:

```
const [mutateUser] = useMyMutation('/api/user')
const [mutatePost] = useMyMutation('/api/post')

mutateUser({ name: 'tom' })
mutatePost({ content: 'foo' })
```

I don't see a way of avoiding calling the hook, which will give you a function that you can invoke eventually (later). But I've also never had the need to...

Or did you have something totally different in mind?

**Reply by anothertempore** [reply]
Yep that's what I'm asking, thanks!
